cmd/cloud-endpoints-controller/main.go

```diff
@@ -126,86 +131,46 @@ func sync(parent *CloudEndpoint, children *CloudEndpointControllerRequestChildre
 	}
 
 	if currState == StateEndpointCreatePending {
+		log.Printf("[INFO] Create pending: %s", parent.Name)
 		var target string
 		var openAPISpec string
 		var err error
+		openAPISpec, err = getUserSuppliedOpenAPISpec(parent, status)
+		//log.Printf(openAPISpec)
+		if err != nil {
+			log.Printf("[INFO] failed to get user supplied spec")
+			return status, &desiredChildren, err
+		}
+		if parent.Spec.TargetIngress.Name != "" {
+			target, status.JWTAudiences, err = getTargetIngress(parent)
+			if err != nil {
+				log.Printf("[INFO] failed ot get target %s", err.Error())
+				return status, &desiredChildren, err
+			}
+		} else {
+			target = parent.Spec.Target
+		}
 
-		if parent.Spec.OpenAPISpec == nil {
+		if openAPISpec == "" {
```

Comment:
Refactor this so that the string version of either the provided spec or the configmap spec is extracted, then executeTemplate and validate once.